src/data_info/services/stat.clj

```diff
@@ -27,6 +29,20 @@
   [stat-map]
   (= (:permission stat-map) :own))
 
+(defn- needs-key?
+  [requested-keys needed-key?]
+  (or (contains? requested-keys needed-key?)
+    (case needed-key?
+      ; :permission is needed by anything which uses `owns?`
+      :permission (contains? requested-keys :share-count)
+      ; :type is needed by anything which uses `is-dir?`
+      :type       (not (empty? (cset/intersection #{:infoType :content-type :file-count :dir-count} requested-keys)))
+      false)))
```

Comment:
One extremely minor suggestion would be to use `(seq)` instead of `(not (empty))`.